File: SDKs/amazonaws.com/kinesisanalytics/2015-08-14/go/pkg/models/shared/kinesisfirehoseinputupdate.go
package shared

// KinesisFirehoseInputUpdate
// When updating application input configuration, provides information about an Amazon Kinesis Firehose delivery stream as the streaming source.
type KinesisFirehoseInputUpdate struct {
	ResourceARNUpdate *string `json:"ResourceARNUpdate,omitempty"`
	RoleARNUpdate     *string `json:"RoleARNUpdate,omitempty"`
}

File: SDKs/amazonaws.com/kinesisanalytics/2015-08-14/java/lib/src/main/java/openapisdk/models/shared/KinesisFirehoseInputUpdate.java
package openapisdk.models.shared;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;

/**
 * KinesisFirehoseInputUpdate
 * When updating application input configuration, provides information about an Amazon Kinesis Firehose delivery stream as the streaming source.
**/
public class KinesisFirehoseInputUpdate {
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("ResourceARNUpdate")
    public String resourceARNUpdate;
    public KinesisFirehoseInputUpdate withResourceArnUpdate(String resourceARNUpdate) {
        this.resourceARNUpdate = resourceARNUpdate;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("RoleARNUpdate")
    public String roleARNUpdate;
    public KinesisFirehoseInputUpdate withRoleArnUpdate(String roleARNUpdate) {
        this.roleARNUpdate = roleARNUpdate;
        return this;
    }
}